controller/commentsController.js

```diff
@@ -405,3 +404,40 @@ exports.getCommentVotes = async (req, res, next) => {
     );
   }
 };
+exports.getSingleComment = async (req, res, next) => {
+  const { refId } = req.query;
+  const commentId = req.params.commentId;
+  const applicationId = req.headers.token; //this will be retrieved from decoded api token after full auth implementation
+  const query = { applicationId: applicationId, _id: commentId };
+  if (refId) query.refId = refId;
+  try {
+    await Comments.find(query)
+      .then((comments) => {
+        const comment = comments.map((comment) => {
+          return {
+            commentId: comment._id,
+            refId: comment.refId,
+            applicationId: comment.applicationId,
+            ownerId: comments.ownerId,
+            content: comment.content,
+            origin: comment.origin,
+            numOfVotes: comment.upVotes.length + comment.downVotes.length,
+            numOfUpVotes: comment.upVotes.length,
+            numOfDownVotes: comment.downVotes.length,
+            numOfFlags: comment.flags.length,
+            numOfReplies: comment.replies.length,
+            // createdAt: comment.createdAt,
+            // updatedAt: comment.updatedAt,
+          };
+        });
+        responseHandler(res, 200, comment, `Comment Retrieved Successfully`);
+      })
+      .catch((err) => {
+        return next(
+          new CustomError(500, "Something went wrong, please try again", err)
```

Comment:
If this is a fail where the record is not found then I think a 404 error code is more appropriate with proper message